pkgs/development/tools/analysis/radare2-cutter/default.nix

```diff
@@ -10,20 +10,45 @@
 
 stdenv.mkDerivation rec {
   name = "radare2-cutter-${version}";
-  version = "1.4";
+  version = "1.5";
 
   src = fetchFromGitHub {
     owner = "radareorg";
     repo = "cutter";
     rev = "v${version}";
-    sha256 = "0wsxb6jfpsmgsigmbnh08j99779bsjz02v6aasqcwl6hwjx0mjfk";
+    sha256 = "0xwls8jhhigdkwyq3nf9xwcz4inm5smwinkyliwmfzvfflbbci5c";
   };
 
   postUnpack = "export sourceRoot=$sourceRoot/src";
 
-  nativeBuildInputs = [ qmake pkgconfig ];
+  # Remove this "very helpful" helper file intended for discovering r2,
+  # as it's a doozy of harddcoded paths and unexpected behavior.
+  # Happily Nix has everything all set so we don't need it,
+  # other than as basis for the qmakeFlags set below.
+  postPatch = ''
+    substituteInPlace Cutter.pro \
+      --replace "include(lib_radare2.pri)" ""
+  '';
+
+  nativeBuildInputs = [ qmake pkgconfig makeWrapper ];
   buildInputs = [ qtbase qtsvg qtwebengine radare2 python3 ];
 
+  qmakeFlags = [
+    "CONFIG+=link_pkgconfig"
+    "PKGCONFIG+=r_core"
+    # Leaving this enabled doesn't break build but generates errors
+    # at runtime (to console) about being unable to load needed bits.
+    # Disable until can be looked at.
+    "CUTTER_ENABLE_JUPYTER=false"
+  ];
+
+  # Fix crash on startup in some situations
+  postInstall = ''
+    wrapProgram $out/bin/Cutter \
+      --prefix QT_PLUGIN_PATH : ${qtbase.bin}/${qtbase.qtPluginPrefix} \
```

Comment:
In nix qt plugins are read from the nix profile. Prefixing the qt that was used for building sounds like a straight-forward idea. However I wonder if it has any side-effects if then other qt components are loaded from the nix profile with a slightly different version (do they have an internal not-stable ABI?)

Reply:
Hmm, very good point.  Fixed here: https://github.com/NixOS/nixpkgs/pull/43425 .  Thanks!